### PR TITLE
[dataframe] Add new port

### PR DESCRIPTION
**Pull request by AlvinZhangH**
dataframe currently only supports Windows, Linux and Mac platforms, not support WindowsStore.
No features need to test.
Related: #3369.

### CODE REVIEW

File: ports/dataframe/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+if(NOT VCPKG_CMAKE_SYSTEM_NAME)
+    vcpkg_from_github(
+        OUT_SOURCE_PATH SOURCE_PATH
+        REPO hosseinmoein/DataFrame
+        REF V-1.5.0
+        SHA512 2eaef420d6b6d7eeeb0e78ec7c9e0bb06d21f8a2c6416c641def567d340b2fdd21f7981df65be2da54117b440afbbc8fa4c7e4d106d954e2b5a17d717322dc02
+        HEAD_REF master
+        PATCHES
+            fix-cmakelists.patch
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake]: Use a macro to distinguish this patch.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/dataframe/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: dataframe
+Version: V-1.5.0
```
Comment (JackBoosY) [inline ports/dataframe/CONTROL]: change to 1.5.0

File: ports/dataframe/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
+include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake]: We don't need include this function now.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
Feel free ask us to reopen this PR if you need it.

**Pull request closed without merging**